lib/data-types.js

```diff
@@ -39,24 +42,47 @@ var util = require('util')
  */
 
 var ABSTRACT = function(options) {
-
 };
 
 ABSTRACT.prototype.dialectTypes = '';
 
-ABSTRACT.prototype.toString = function() {
-  return this.toSql();
+ABSTRACT.prototype.toString = function(options) {
+  return this.toSql(options);
 };
 ABSTRACT.prototype.toSql = function() {
   return this.key;
 };
-ABSTRACT.prototype.warn = function(text) {
-  if (!warnings[text]) {
-    warnings[text] = true;
-    console.warn('>> WARNING:', text, '\n>> Check:', this.dialectTypes);
+ABSTRACT.warn = function(link, text) {
+  deprecate(text + '\n>> Check: ' + link);
```

Comment:
If I'm not mistaken the issue with the call stack is that the warning is on the prototype and called from a internal function. That way depd is going to look at the next upper level and find the internal function that called it and prints that as the wrong line. So the depd call would have to happen in the function itself. It might work if you do something like that (I'm not sure though):
```js
ABSTRACT.warn = function (text) {
  return text + '\n>> Check: ' + this.dialectTypes;
};
// And now somehwere where the warning is triggered:
function trigger (value) {
  if (typeof value !== 'number') {
    deprecate(this.warn(text));
  }
}
```
But it would not solve the fact that it's not really a deprecation :) (and you can't suppress that word). If you want the colors that's easy to implement without the library. Just use something like `console.log('\x1b[36mCOLORED_WORDS\x1b[0m');`